src/solver/mha/mha_ck_solver_forward.cpp

```diff
@@ -0,0 +1,323 @@
+    elementwise_bias = 1,
+    alibi            = 2,
+};
+
+struct fmha_fwd_args
+{
+    const void* q_ptr;
+    const void* k_ptr;
+    const void* v_ptr;
+    const void* bias_ptr; // bias or alibi_slope pointer
+    void* lse_ptr;
+    void* o_ptr;
+    const void* seqstart_q_ptr;
+    const void* seqstart_k_ptr;
+    const void* seqlen_k_ptr;
+    ck_tile::index_t seqlen_q;
+    ck_tile::index_t seqlen_k;
+    ck_tile::index_t batch;
+    ck_tile::index_t max_seqlen_q;
+    ck_tile::index_t hdim_q;
+    ck_tile::index_t hdim_v;
+    ck_tile::index_t nhead_q;
+    ck_tile::index_t nhead_k;
+    float scale_s;
+    float scale_p;
+    float scale_o;
+    ck_tile::index_t stride_q;
+    ck_tile::index_t stride_k;
+    ck_tile::index_t stride_v;
+    ck_tile::index_t stride_bias; // if alibi, b*h need set this to h, 1*h need set this to 0
+    ck_tile::index_t stride_o;
+    ck_tile::index_t nhead_stride_q;
+    ck_tile::index_t nhead_stride_k;
+    ck_tile::index_t nhead_stride_v;
+    ck_tile::index_t nhead_stride_bias;
+    ck_tile::index_t nhead_stride_lse;
+    ck_tile::index_t nhead_stride_o;
+    ck_tile::index_t batch_stride_q;
+    ck_tile::index_t batch_stride_k;
+    ck_tile::index_t batch_stride_v;
+    ck_tile::index_t batch_stride_bias;
+    ck_tile::index_t batch_stride_lse;
+    ck_tile::index_t batch_stride_o;
+    ck_tile::index_t window_size_left;
+    ck_tile::index_t window_size_right;
+    ck_tile::index_t mask_type;
+};
+
+struct stream_config
+{
+    hipStream_t stream_id_ = nullptr;
+    bool time_kernel_      = false;
+    int log_level_         = 0;
+    int cold_niters_       = 3;
+    int nrepeat_           = 10;
+};
+
+struct fmha_fwd_traits
+{
+    int hdim_q;
+    int hdim_v;
+    std::string data_type;
+    bool is_group_mode;
+    bool is_v_rowmajor;
+    mask_enum mask_type;
+    bias_enum bias_type; // 0:no bias, 1:elementwise bias, 2:alibi. sync with BlockAttentionBiasEnum
+    bool has_lse;
+    bool do_fp8_static_quant;
+    // TODO: padding check is inside this api
+};
+float fmha_fwd(fmha_fwd_traits, fmha_fwd_args, const stream_config&);
+
+bool MhaCKForward::IsApplicable([[maybe_unused]] const ExecutionContext& context,
+                                const miopen::mha::ProblemDescription& problem) const
+{
+    return true;
+}
+
+std::size_t MhaCKForward::GetWorkspaceSize([[maybe_unused]] const ExecutionContext& context,
+                                           const miopen::mha::ProblemDescription& problem) const
+{
+    return 0;
+}
+
+ConvSolution MhaCKForward::GetSolution(const ExecutionContext& context,
+                                       const miopen::mha::ProblemDescription& problem) const
+{
+    auto result         = ConvSolution{miopenStatusSuccess};
+    result.workspace_sz = 0;
+
+    const miopen::mha::MhaInputDescsForward& descsFwd = problem.GetDescsForward();
+
+    uint64_t N, H, S, D;
+    std::tie(N, H, S, D) = miopen::tien<4>(descsFwd.kDesc.GetLengths());
```

Comment:
If you don't need to capture these values it can be just a structure binding:
```suggestion
    auto [N, H, S, D] = miopen::tien<4>(descsFwd.kDesc.GetLengths());
```